File: app/src/main/java/com/app/syspoint/ui/cobranza/AdapterListaDocumentosCobranza.java
package com.app.syspoint.ui.cobranza;

import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ImageView;
import android.widget.TextView;

import androidx.annotation.NonNull;
import androidx.recyclerview.widget.RecyclerView;

import com.app.syspoint.R;
import com.app.syspoint.repository.database.bean.CobranzaBean;
import com.app.syspoint.utils.Utils;

import java.util.List;

public class AdapterListaDocumentosCobranza extends RecyclerView.Adapter<AdapterListaDocumentosCobranza.Holder> {

    private List<CobranzaBean> mData;
    private OnItemClickListener itemClickListener;
    private OnItemLongClickListener onItemLongClickListener;



    public AdapterListaDocumentosCobranza(List<CobranzaBean> mData, OnItemClickListener itemClickListener, OnItemLongClickListener onItemLongClickListener) {
        this.mData = mData;
        this.itemClickListener = itemClickListener;
        this.onItemLongClickListener = onItemLongClickListener;
    }

    @NonNull
    @Override
    public Holder onCreateViewHolder(@NonNull ViewGroup parent, int viewType) {
        View v = LayoutInflater.from(parent.getContext())
                .inflate(R.layout.item_lista_documentos_cardview, parent, false);

        return new Holder(v);
    }

    @Override
    public void onBindViewHolder(@NonNull Holder holder, int position) {
        holder.bind(mData.get(position), itemClickListener,onItemLongClickListener);

    }

    @Override
    public int getItemCount() {
        if (mData.isEmpty()){
            return 0;
        }else {
            return this.mData.size();
        }
    }
    public void setData(List<CobranzaBean>data){
        this.mData = data;
        notifyDataSetChanged();
    }


    public class Holder extends RecyclerView.ViewHolder {

        private TextView textViewVenta;
        private TextView textViewCobranza;
        private TextView textViewImporte;
        private TextView textViewSaldo;
        private TextView textViewFecha;
        private TextView textViewFechaVence;
        private ImageView imageView;

        public Holder(@NonNull View itemView) {
            super(itemView);
            this.textViewVenta = itemView.findViewById(R.id.tv_venta_cobranza_view);
            this.textViewCobranza = itemView.findViewById(R.id.tv_cobranza_cobranza_view);
            this.textViewImporte = itemView.findViewById(R.id.tv_importe_cobranza_view);
            this.textViewSaldo = itemView.findViewById(R.id.tv_saldo_cobranza_view);
            this.textViewFecha = itemView.findViewById(R.id.tv_fecha_cobranza_view);
            this.imageView = itemView.findViewById(R.id.imageView_check);
        }

        public void bind(final CobranzaBean cobranzaBean, final OnItemClickListener listener, final OnItemLongClickListener onItemLongClickListener){

            this.textViewVenta.setText(""+cobranzaBean.getVenta());
            this.textViewCobranza.setText(""+cobranzaBean.getCobranza());
            this.textViewImporte.setText(Utils.FDinero(cobranzaBean.getImporte()));
            this.textViewSaldo.setText(Utils.FDinero(cobranzaBean.getSaldo()));
            this.textViewFecha.setText(cobranzaBean.getFecha());
            if (cobranzaBean.getIsCheck()){
                imageView.setVisibility(View.VISIBLE);
            }else {
                imageView.setVisibility(View.GONE);
            }

            itemView.setOnClickListener(new View.OnClickListener() {
                @Override
                public void onClick(View view) {
                    listener.onItemClick(getAdapterPosition());
                }
            });

            itemView.setOnLongClickListener(new View.OnLongClickListener() {
                @Override
                public boolean onLongClick(View v) {
                    onItemLongClickListener.onItemLongClicked(getAdapterPosition());
                    return false;
                }
            });
        }
    }

    public interface OnItemClickListener{
        void onItemClick(int position);
    }

    public interface OnItemLongClickListener {
        boolean onItemLongClicked(int position);
    }

}

File: app/src/main/java/com/app/syspoint/ui/cobranza/adapter/AdapterListaDocumentosCobranza.kt
package com.app.syspoint.ui.cobranza.adapter

import android.view.LayoutInflater
import android.view.ViewGroup
import androidx.recyclerview.widget.RecyclerView
import com.app.syspoint.databinding.ItemListaDocumentosCardviewBinding
import com.app.syspoint.repository.database.bean.CobranzaBean
import com.app.syspoint.utils.Utils
import com.app.syspoint.utils.setInvisible
import com.app.syspoint.utils.setVisible

class AdapterListaDocumentosCobranza(
    data: List<CobranzaBean?>,
    val onItemClickListener: OnItemClickListener,
    val onItemLongClickListener: OnItemLongClickListener
    ): RecyclerView.Adapter<AdapterListaDocumentosCobranza.Holder>() {

    private var mData = data

    interface OnItemClickListener {
        fun onItemClick(position: Int)
    }

    interface OnItemLongClickListener {
        fun onItemLongClicked(position: Int): Boolean
    }

    override fun onCreateViewHolder(
        parent: ViewGroup,
        viewType: Int
    ): Holder {
        val binding = ItemListaDocumentosCardviewBinding.inflate(LayoutInflater.from(parent.context), parent, false)
        return Holder(binding)
    }

    override fun onBindViewHolder(holder: Holder, position: Int) {
        holder.bind(mData[position], onItemClickListener, onItemLongClickListener)
    }

    override fun getItemCount(): Int= if (mData.isEmpty()) 0 else mData.size

    fun setData(data: List<CobranzaBean?>) {
        mData = data
        notifyDataSetChanged()
    }

    class Holder(val binding: ItemListaDocumentosCardviewBinding): RecyclerView.ViewHolder(binding.root) {

        fun bind(cobranzaBean: CobranzaBean?, onItemClickListener: OnItemClickListener, onItemLongClickListener: OnItemLongClickListener) {
            cobranzaBean?.let {
                binding.tvVentaCobranzaView.text = cobranzaBean.venta.toString()
                binding.tvCobranzaCobranzaView.text = cobranzaBean.cobranza
                binding.tvImporteCobranzaView.text = Utils.FDinero(cobranzaBean.importe)
                binding.tvSaldoCobranzaView.text = Utils.FDinero(cobranzaBean.saldo)
                binding.tvFechaCobranzaView.text = cobranzaBean.fecha

                if (cobranzaBean.isCheck) {
                    binding.imageViewCheck.setVisible()
                } else {
                    binding.imageViewCheck.setInvisible()
                }

                itemView.setOnClickListener { onItemClickListener.onItemClick(adapterPosition) }

                itemView.setOnLongClickListener {
                    onItemLongClickListener.onItemLongClicked(adapterPosition)
                    false
                }
            }
        }
    }
}